fix(core): allow passing value of any type to `isSignal` function

markostanimirovic opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

To check if a value of type `unknown` is a Signal, we need to add two conditions:

```ts
import { isSignal } from '@angular/core';

declare const potentialSignal: unknown;

const isSig = typeof potentialSignal === 'function' && isSignal(potentialSignal);
```

## What is the new behavior?

The first condition is not needed anymore, so we can call the `isSignal` function and pass the argument of type `unknown`:

```ts
import { isSignal } from '@angular/core';

declare const potentialSignal: unknown;

const isSig = isSignal(potentialSignal);
```

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

`isObservable` type guard from the `rxjs` package accepts value of any type as an input argument.

crisbeto reviewed (approved):
LGTM

crisbeto commented:
I just wanted to note that we're currently in a feature freeze which doesn't allow new `feat` commits until v16 is released. That being said, I think that this is more of a fix than a feature.

markostanimirovic committed:
fix(core): allow passing value of any type to `isSignal` function

Unlike the current signature where the input argument must a function, this change allows an input of any type to be passed to the `isSignal` function.

pkozlowski-opensource commented:
Marking as cleanup to add tests and update the commit message to the `fix`

pkozlowski-opensource reviewed (approved):
LGTM

Reviewed-for: public-api 
Reviewed-for: fw-core

jessicajaniuk reviewed (approved):
reviewed-for: public-api

AndrewKushnir commented:
This PR was merged into the repository by commit 165b8b647c39503b9fefd00188948cdc522c2c10.
